### PR TITLE
Bani-- Customer Portal

### PR DESCRIPTION
**Pull request by Bani2791**
# 1. CP>Dashboard>Internal number block

## Internal Number Block

Internal Number Block to define the range of numbers a Customer can use for setting up SIP Extensions.

### Why Define Internal Number Blocks?

Within each Account are multiple Customers, each will likely have more than one Contact.

SIP Extensions are unique for the Account, not for individual Customers.

Without defined Internal Number Blocks, there is the risk of a Contact attempting to use an extension that's already in use by another Customer or Contact.
With Internal Number Blocks defined, each Customer will have a predetermined list of Usernames to select for SIP Authentication.
[Dashboard - ConnexCS Documentation (bani-customerportal--connexcs-docs.netlify.app)](https://bani-customerportal--connexcs-docs.netlify.app/customer-portal/cp-dashboard/#internal-number-block)

# 2. CP> CDR

!!! Info

    When a call enters the CDR table, you have the option to **Play Audio**. The **Call recording** is available, and you can **Download** it as well.

[CDR - ConnexCS Documentation (bani-customerportal--connexcs-docs.netlify.app)](https://bani-customerportal--connexcs-docs.netlify.app/customer-portal/cp-cdr/)

# 3. CP>Breakout

 The **Select Destination** allows you to filter out the calls made to a particular destination.

|**Total Provider Charge**|How much your provider is charging you|

|**Total Customer Charge**|How much you are charging your customer|

[Breakout - ConnexCS Documentation (bani-customerportal--connexcs-docs.netlify.app)](https://bani-customerportal--connexcs-docs.netlify.app/customer-portal/cp-breakout/)

# 4. CP>DID

1. For single DID, click on number, click on **Edit** (beside Destination) and you will see the following fields in a new window.

    + **URI:** Set the **Destination DID** (number or extension) and **Destination IP** to forward calls to, or a specific Session Initiation Protocol (SIP) Uniform Resource Identifier (URI)  (formatted without the initial sip:, for example, 1-999-123-4567@voip-provider.example.net)

    + **External:** Prefix sends the call to the internet and then the number to send the call to.

    + **Internal:** Send internally to an extension, a Class5 feature, or even to another customer.

    + **Circuit Test**: *in progress*

3. Click **Save**.

You have another filed here, **Tags**, it's a **label** which helps you to describe and help organize information.

4.For multiple DIDs, select multiple DIDs, then click on **Bulk Edit**.

[5.Click](http://5.click/) on **Edit** (beside Destination)

6.Enter the values for various fields like **URI**, **External**, or **Internal**.

[DID - ConnexCS Documentation (bani-customerportal--connexcs-docs.netlify.app)](https://bani-customerportal--connexcs-docs.netlify.app/customer-portal/cp-did/)

# 5. CP> CLI

## Steps to add Approved CLIs

1. Login to your account.

2. Click on **CLI** on the left panel.

3. Click on the blue + button on the top right.

4. Enter the phone number with the country prefix.

5. Click on **Next Step**.

[6.You](http://6.you/) will receive a call for the **Verification Code**.

7.Enter the verification code and the click **Verify Code**.

8.The CLI will be added as **Whitelist**. **Forced** CLI appears when the customer has forced a particular CLI in their account.

[DID - ConnexCS Documentation (bani-customerportal--connexcs-docs.netlify.app)](https://bani-customerportal--connexcs-docs.netlify.app/customer-portal/cp-cli/)

# 7. CP>Documents

You can also view the documents by just clicking on the uploaded document under **File Name** column.

[Documents - ConnexCS Documentation (bani-customerportal--connexcs-docs.netlify.app)](https://bani-customerportal--connexcs-docs.netlify.app/customer-portal/cp-documents/)

# 8. CP>Dashboard

### Setup Personal Profile

To setup, click your name in the upper right corner and select My Profile.

=== "Basic"

    View account details such as **Name**, **Role**, **Email**, **Department**, **Phone**,  **Mobile** and **Skype**. 

    **You can only edit the Mobile field. To change email or name, contact ConnexCS Support.**

    **Two Factor Authentication**: This adds additional security against password guessing and theft, including brute force attacks (details [**here**](https://www.ncsc.gov.uk/guidance/multi-factor-authentication-online-services)). 

    To configure this on ConnexCS, click Enable and follow prompts for QR code. You must install a 2FA/TFA app on your phone. 

    

https://bani-customerportal--connexcs-docs.netlify.app/customer-portal/cp-dashboard/

# 9. CP>Routes

You can also view the **Rate Card** associated with Termination routes.

[Routes - ConnexCS Documentation (bani-customerportal--connexcs-docs.netlify.app)](https://bani-customerportal--connexcs-docs.netlify.app/customer-portal/cp-routes/)

# 10. CP>Package

+ **Frequency:** You can access the package either monthly, daily, weekly.

+ **Auto Decrement Credit:** Selecting Auto Decrement Credit will debit the package amount from the customer's account.

[Package - ConnexCS Documentation (bani-customerportal--connexcs-docs.netlify.app)](https://bani-customerportal--connexcs-docs.netlify.app/customer-portal/cp-package/)

# 11. CP>Documents

You can also view the documents by just clicking on the uploaded document under **File Name** column.

[Documents - ConnexCS Documentation (bani-customerportal--connexcs-docs.netlify.app)](https://bani-customerportal--connexcs-docs.netlify.app/customer-portal/cp-documents/)

# 12. CP>Data-management

You can view the databases (only which are assigned to a particular customer) here created on the **Control Panel**.

Also, clicking on the blue + sign you can add new **Key-Value** pair.

[Data Management - ConnexCS Documentation (bani-customerportal--connexcs-docs.netlify.app)](https://bani-customerportal--connexcs-docs.netlify.app/customer-portal/cp-data-management/)

### CODE REVIEW

File: docs/customer-portal/cp-dashboard.md [hunk 1]
```diff
@@ -41,3 +57,17 @@ You can perform the following Functions with this data:
...
 
 [cp-dash]: /customer-portal/img/cp-dash.png "Top of Page"
+
+## Internal Number Block
+
+Internal Number Block to define the range of numbers a Customer can use for setting up SIP Extensions.
+
+### Why Define Internal Number Blocks?
+
+Within each Account are multiple Customers, each will likely have more than one Contact.
```
Comment (AdamNawaz-connex) [inline docs/customer-portal/cp-dashboard.md:67]: No need for extra information on Internal Number Block as its Customer Portal documentation 
Just define what is internal Number Block and they can see what range has been assigned to them

File: docs/customer-portal/cp-cdr.md [hunk 2]
```diff
@@ -24,3 +24,6 @@ Coordinated Universal Time (UTC) rates and stores the CDRs.
 Daily totals are also calculated in UTC.
 
 You can change the display time zone for each CDR record and saves the time zone selector, but downloads will always be in UTC.
+
+!!! Info
+    When a call enters the CDR table, you have the option to **Play Audio**. The **Call recording** is available, and you can **Download** it as well.
```
Comment (AdamNawaz-connex) [inline docs/customer-portal/cp-cdr.md:29]: If a recording of the call is present, you have the option to play the recording and download it as well.

File: docs/customer-portal/cp-did.md [hunk 3]
```diff
@@ -5,8 +5,27 @@ A **DID** number is one that exists on the public telephone network. When dialed
 To edit **DID (Direct Inward Dial)**:
 
 1. Select one DID by clicking on the number, or select multiple and then click **`Bulk Edit`**.
+2. For single DID, click on number, click on **Edit** (beside Destination) and you will see the following fields in a new window.
```
Comment (AdamNawaz-connex) [inline docs/customer-portal/cp-did.md:8]: No need for "For single DID, click on number, "
As its already explained in step1

File: docs/customer-portal/cp-did.md [hunk 4]
```diff
@@ -5,8 +5,27 @@ A **DID** number is one that exists on the public telephone network. When dialed
 To edit **DID (Direct Inward Dial)**:
 
 1. Select one DID by clicking on the number, or select multiple and then click **`Bulk Edit`**.
+2. For single DID, click on number, click on **Edit** (beside Destination) and you will see the following fields in a new window.
+
     + **URI:** Set the **Destination DID** (number or extension) and **Destination IP** to forward calls to, or a specific Session Initiation Protocol (SIP) Uniform Resource Identifier (URI)  (formatted without the initial `sip:`, for example, `1-999-123-4567@voip-provider.example.net`)
```
Comment (AdamNawaz-connex) [inline docs/customer-portal/cp-did.md:10]: Set the Destination DID (number or extension) and IP to forward calls to a specific SIP URI
This is enough

File: docs/customer-portal/cp-did.md [hunk 5]
```diff
@@ -5,8 +5,27 @@ A **DID** number is one that exists on the public telephone network. When dialed
 To edit **DID (Direct Inward Dial)**:
 
 1. Select one DID by clicking on the number, or select multiple and then click **`Bulk Edit`**.
+2. For single DID, click on number, click on **Edit** (beside Destination) and you will see the following fields in a new window.
+
     + **URI:** Set the **Destination DID** (number or extension) and **Destination IP** to forward calls to, or a specific Session Initiation Protocol (SIP) Uniform Resource Identifier (URI)  (formatted without the initial `sip:`, for example, `1-999-123-4567@voip-provider.example.net`)
     + **External:** Prefix sends the call to the internet and then the number to send the call to.
```
Comment (AdamNawaz-connex) [inline docs/customer-portal/cp-did.md:11]: To send the call back out to the internet

File: docs/customer-portal/cp-did.md [hunk 6]
```diff
@@ -5,8 +5,27 @@ A **DID** number is one that exists on the public telephone network. When dialed
 To edit **DID (Direct Inward Dial)**:
 
 1. Select one DID by clicking on the number, or select multiple and then click **`Bulk Edit`**.
+2. For single DID, click on number, click on **Edit** (beside Destination) and you will see the following fields in a new window.
+
     + **URI:** Set the **Destination DID** (number or extension) and **Destination IP** to forward calls to, or a specific Session Initiation Protocol (SIP) Uniform Resource Identifier (URI)  (formatted without the initial `sip:`, for example, `1-999-123-4567@voip-provider.example.net`)
     + **External:** Prefix sends the call to the internet and then the number to send the call to.
     + **Internal:** Send internally to an extension, a Class5 feature, or even to another customer.
-    + **Circuit Test**: *in progress*
```
Comment (AdamNawaz-connex) [inline docs/customer-portal/cp-did.md:11]: It's not available on CP

File: docs/customer-portal/cp-cli.md [hunk 7]
```diff
@@ -21,6 +21,21 @@ Depending on how you want to route the calls, toggle a CLI between `Forced` and
...
+5. Click on **Next Step**.
+
+<img src= "docs/customer-portal/img/cli1.png">
+
+6.You will receive a call for the **Verification Code**.
+7.Enter the verification code and the click **Verify Code**.
+
+<img src= "docs/customer-portal/img/cli2.png">
+
+8.The CLI will be added as **Whitelist**. **Forced** CLI appears when the customer has forced a particular CLI in their account.
```
Comment (AdamNawaz-connex) [inline docs/customer-portal/cp-cli.md:39]: Add **Forced** CLI appears when the customer has forced a particular CLI in their account.  as a note

File: docs/customer-portal/cp-cli.md [hunk 8]
```diff
@@ -21,6 +21,21 @@ Depending on how you want to route the calls, toggle a CLI between `Forced` and
...
--->
+## Steps to add Approved CLIs
+
+1. Login to your account.
+2. Click on **CLI** on the left panel.
+3. Click on the `blue +` button on the top right.
+4. Enter the phone number with the country prefix.
+5. Click on **Next Step**.
+
+<img src= "docs/customer-portal/img/cli1.png">
```
Comment (AdamNawaz-connex) [inline docs/customer-portal/cp-cli.md:32]: These images are not visible

File: docs/customer-portal/cp-documents.md [hunk 9]
```diff
@@ -1,3 +1,5 @@
 # Documents
 
 View all **Documents** associated to your account. This could include feature lists or payment requirements.
+
+You can also view the documents by just clicking on the uploaded document under **File Name** column.
```
Comment (AdamNawaz-connex) [inline docs/customer-portal/cp-documents.md:5]: You can also view the documents by clicking on the uploaded document in the File Name column.

File: docs/customer-portal/cp-dashboard.md [hunk 10]
```diff
@@ -12,20 +12,36 @@ The **Customer Portal** provides access to configurations, real-time testing and
...
 
+### Setup Personal Profile
+
+To setup, click your name in the upper right corner and select `My Profile`.
+
+=== "Basic"
+
+    View account details such as **Name**, **Role**, **Email**, **Department**, **Phone**,  **Mobile** and **Skype**. 
+    
+    **You can only edit the Mobile field. To change email or name, contact ConnexCS Support.**
```
Comment (AdamNawaz-connex) [inline docs/customer-portal/cp-dashboard.md:27]: They can Change Name  Phone, Mobile ,Skype

File: docs/customer-portal/cp-dashboard.md [hunk 11]
```diff
@@ -12,20 +12,36 @@ The **Customer Portal** provides access to configurations, real-time testing and
...
-+ **Account Balance** Displays the current account balance and a link to make a payment using a credit card or PayPal.
++ **Account Balance** Displays the current account balance and a link to make a payment using a credit card or PayPal. [Click here](https://docs.connexcs.com/customer-portal/cp-payment/) to know more about **Payments**.
 
 + **Account** Use this to view your Company Profile, view and update your personal Profile, Change Password or Language, or Log Out.
 
+### Setup Personal Profile
+
+To setup, click your name in the upper right corner and select `My Profile`.
+
+=== "Basic"
```
Comment (AdamNawaz-connex) [inline docs/customer-portal/cp-dashboard.md:23]: No need for this

File: docs/customer-portal/cp-dashboard.md [hunk 12]
```diff
@@ -12,20 +12,36 @@ The **Customer Portal** provides access to configurations, real-time testing and
...
+
+    View account details such as **Name**, **Role**, **Email**, **Department**, **Phone**,  **Mobile** and **Skype**.
```
Comment (AdamNawaz-connex) [inline docs/customer-portal/cp-dashboard.md:25]: No need to Capitalize them

File: docs/customer-portal/cp-dashboard.md [hunk 13]
```diff
@@ -12,20 +12,36 @@ The **Customer Portal** provides access to configurations, real-time testing and
...
 
 Below is the Current Customer account usage on four default **Graphs**.
 
 |Graph|Description |
-|:------------|:-------------------------------------------------|
-|**Live Channels** |Details of active channel usage|
-|**ASR** |You can view Answer Seizure Ratio as a Total, or isolate traffic on a per-carrier basis (click on a customer to remove or add it from the graph); if some carriers are lower than others, this may create an issue|
+|:----|:-----------|
+|**Live Channels**|Details of active channel usage|
+|**ASR**|You can view Answer Seizure Ratio as a Total, or isolate traffic on a per-carrier basis (click on a customer to remove or add it from the graph); if some carriers are lower than others, this may create an issue|
```
Comment (AdamNawaz-connex) [inline docs/customer-portal/cp-dashboard.md:42]: We cannot customize these graphs

File: docs/customer-portal/cp-dashboard.md [hunk 14]
```diff
@@ -12,20 +12,36 @@ The **Customer Portal** provides access to configurations, real-time testing and
...
 |**ACD**|You can view the total average call duration or isolate traffic by carrier (click on a customer to remove or add it to the graph). A lower ACD on one carrier is indicative of potential line quality issues|
```
Comment (AdamNawaz-connex) [inline docs/customer-portal/cp-dashboard.md:43]: We cannot customize these graphs

File: docs/customer-portal/cp-dashboard.md [hunk 15]
```diff
@@ -12,20 +12,36 @@ The **Customer Portal** provides access to configurations, real-time testing and
...
-|**DTMF** |DTMF (Dual Tone Modulated Frequency) percentages|
+|**DTMF**|DTMF (Dual Tone Modulated Frequency) percentages|
 
 ## Summary
```
Comment (AdamNawaz-connex) [inline docs/customer-portal/cp-dashboard.md:46]: In summary remove Generate Invoice as its CP and we don't have an option to Generate Invoice in CP

File: docs/customer-portal/cp-routes.md [hunk 16]
```diff
@@ -1,6 +1,6 @@
 # Routes
 
-All **Termination Routes**, or outbound calls.
+All **Termination Routes**, or outbound calls. You can also view the **Rate Card** associated with Termination routes.
```
Comment (AdamNawaz-connex) [inline docs/customer-portal/cp-routes.md:3]: All Termination Routes can be viewed. You can also access the Rate Card associated with Termination routes."

File: docs/customer-portal/cp-package.md [hunk 17]
```diff
@@ -9,3 +9,5 @@ Get a quick reference view of your products and services:
 + **Name:** The name of the package.
 + **Retail:** Recurring charges and frequency (for example, monthly), any one-time charges.
 + **Description:** Product description.
++ **Frequency:** You can access the package either monthly, daily, weekly.
```
Comment (AdamNawaz-connex) [inline docs/customer-portal/cp-package.md:12]: when the package charge repeats

File: docs/customer-portal/cp-data-management.md [hunk 18]
```diff
@@ -4,27 +4,6 @@ The **Database** allows you to create custom persistent data storage. A datastor
```
Comment (AdamNawaz-connex) [inline docs/customer-portal/cp-data-management.md:4]: A datastore can be specific to a single customer or global (used with any customer).
Is that the beginning please remove that

File: docs/customer-portal/cp-dialogs.md [hunk 19]
```diff
@@ -6,7 +6,5 @@
 
 !!! note "Ended Status"
     "Ended" indicates a call is over, but the telephony switches are still tearing down the call, including the Real-time Transport Protocol (RTP) stream and any final signaling. After that, the server gathers data for billing.
-!!! note "Global Dialogs"
-    You can also see the status of all the calls for specific Customers in **Global :material-menu-right: Dialogs**.
 
-    Also, you can select and end the calls using **End Calls** and the calls are terminaed using the MI Termination.
+    Also, you can select and end the calls using **End Calls** and the calls are terminated using the MI Termination.
```
Comment (AdamNawaz-connex) [inline docs/customer-portal/cp-dialogs.md:10]: They cannot end calls from CP 
Remove this part

File: docs/customer-portal/cp-dashboard.md (hunk 1 above)
Comment (Bani2791) [reply docs/customer-portal/cp-dashboard.md:67]: removed extra information

File: docs/customer-portal/cp-cdr.md (hunk 2 above)
Comment (Bani2791) [reply docs/customer-portal/cp-cdr.md:29]: re-written the statement

File: docs/customer-portal/cp-did.md (hunk 4 above)
Comment (Bani2791) [reply docs/customer-portal/cp-did.md:10]: removed the example

File: docs/customer-portal/cp-did.md (hunk 5 above)
Comment (Bani2791) [reply docs/customer-portal/cp-did.md:11]: made the correction

File: docs/customer-portal/cp-did.md (hunk 6 above)
Comment (Bani2791) [reply docs/customer-portal/cp-did.md:11]: removed circuit test

File: docs/customer-portal/cp-cli.md (hunk 7 above)
Comment (Bani2791) [reply docs/customer-portal/cp-cli.md:39]: added as a note

File: docs/customer-portal/cp-documents.md (hunk 9 above)
Comment (Bani2791) [reply docs/customer-portal/cp-documents.md:5]: corrected

File: docs/customer-portal/cp-dashboard.md (hunk 10 above)
Comment (Bani2791) [reply docs/customer-portal/cp-dashboard.md:27]: corrected

File: docs/customer-portal/cp-dashboard.md (hunk 11 above)
Comment (Bani2791) [reply docs/customer-portal/cp-dashboard.md:23]: ask adam

File: docs/customer-portal/cp-dashboard.md (hunk 12 above)
Comment (Bani2791) [reply docs/customer-portal/cp-dashboard.md:25]: ask adam

File: docs/customer-portal/cp-dashboard.md (hunk 14 above)
Comment (Bani2791) [reply docs/customer-portal/cp-dashboard.md:43]: ask adam

File: docs/customer-portal/cp-dashboard.md (hunk 15 above)
Comment (Bani2791) [reply docs/customer-portal/cp-dashboard.md:46]: removed

File: docs/customer-portal/cp-routes.md (hunk 16 above)
Comment (Bani2791) [reply docs/customer-portal/cp-routes.md:3]: corrected

File: docs/customer-portal/cp-package.md (hunk 17 above)
Comment (Bani2791) [reply docs/customer-portal/cp-package.md:12]: ask adam

File: docs/customer-portal/cp-dialogs.md (hunk 19 above)
Comment (Bani2791) [reply docs/customer-portal/cp-dialogs.md:10]: ask adam

File: docs/customer-portal/cp-cli.md (hunk 8 above)
Comment (Bani2791) [reply docs/customer-portal/cp-cli.md]: rectified

**Review by AdamNawaz-connex — APPROVED** [review]

**Pull request merged**